### ISSUE TITLE
Java CI Job not running any tests

### ISSUE THREAD

**Issue by StruckCroissant** [StruckCroissant/Game-DB#38]
As seen in [Java CI #3](https://github.com/StruckCroissant/Game-DB/actions/runs/2872712812) - the job is currently not running tests. Previously, there were issues with getting this job to run at all, which I could fix by adding `<start-class>` line to `pom.xml`. 
```xml
<properties>
	<java.version>17</java.version>
	<start-class>com.StruckCroissant.GameDB.GameDbApplication</start-class>
</properties>
```

It looks like this issue could be because of a missing `<testSourceDirectory>` tag. For more details, see these Stack Overflow issues:
https://stackoverflow.com/questions/66907650/github-ci-with-maven-testing-not-running-any-tests
https://stackoverflow.com/questions/19139509/maven-project-build-testsourcedirectory-property-not-working-from-profile

**Comment by StruckCroissant** [comment]
adding `testSourceDirectory` did not resolve the issue.

**Comment by StruckCroissant** [comment]
Refactored Java source folder from `api` to `src`, & that 'fixed' the issue (aka there's just a different error now that shows me that the job actually sees the tests). 
Refactor was done in commit 144a257 - going to revert from this commit to previous commit for now because I would like to avoid a large refactor like that in favor of just overwriting the source in the pom.

[see here for 'successful' build execution](https://github.com/StruckCroissant/Game-DB/runs/7891574550?check_suite_focus=true)

**Comment by StruckCroissant** [comment]
Reverted refactor in 69cea3a

**Comment by StruckCroissant** [comment]
fixed in ade00cd - the issue was that the there were no source/test directories in the pom.

### Additions:
```xml
<sourceDirectory>
			${project.basedir}/api/main/java
		</sourceDirectory>
		<testSourceDirectory>
			${project.basedir}/api/test/java
		</testSourceDirectory>
		<resources>
			<resource>
				<directory>
					${project.basedir}/api/main/resources
				</directory>
			</resource>
		</resources>
		<testResources>
			<testResource>
				<directory>
					${project.basedir}/api/test/resources
				</directory>
			</testResource>
		</testResources>
</sourceDirectory>
```

### Successful job run:
[Java CI #16](https://github.com/StruckCroissant/Game-DB/runs/7923705968?check_suite_focus=true)
